File: home/dot_local/lib/theme-gallery/samples/sample.go
// Palette sample: Go — comments, strings, numbers, keywords, types, funcs.
package main

import (
	"fmt"
	"strings"
)

const maxRetries = 3

// Greeter builds greetings.
type Greeter struct {
	Name string
}

func (g Greeter) Greet() string {
	if g.Name == "" {
		g.Name = "world"
	}
	return fmt.Sprintf("hello %s\n", strings.TrimSpace(g.Name))
}

func main() {
	for i := 0; i < maxRetries; i++ {
		fmt.Print(Greeter{Name: "palette"}.Greet())
	}
}

File: home/dot_local/lib/theme-gallery/samples/sample.cs
// Palette sample: C# — comments, strings, numbers, keywords, types, xml-doc.
using System;

namespace Palette;

/// <summary>Builds greetings for the palette demo.</summary>
public sealed class Greeter(string name = "world")
{
    private const int MaxRetries = 3;

    public string Greet()
    {
        var sb = new System.Text.StringBuilder();
        for (var i = 0; i < MaxRetries; i++)
        {
            sb.AppendLine($"hello {name} ({i})");
        }
        return sb.ToString();
    }
}
